lib/ecmascript.ts

```diff
@@ -2266,7 +2266,7 @@ export const ES = ObjectAssign({}, ES2020, {
     return ES.BalanceISODateTime(year, month, day, hour, minute, second, millisecond, microsecond, nanosecond);
   },
   GetIANATimeZoneNextTransition: (epochNanoseconds, id) => {
-    const uppercap = ES.SystemUTCEpochNanoSeconds() + 366 * DAYMILLIS * 1e6;
+    const uppercap = ES.SystemUTCEpochNanoSeconds().plus(bigInt(366).multiply(DAYMILLIS).multiply(1e6));
```

Comment:
Minor nit: we don't need bigint math until it goes above 104 days, so no need for two `multiply` calls here.  Only one is needed.

That said, everywhere DAYMILLIS is used, it's multiplied by 1e6 and used as a bigint. So it may make sense as part of this PR to replace DAYMILLIS with `const DAYNANOS = bigInt(86400).multiply(1e9);`.  Could also put our use of 366 days and 14 days into bigint constants instead of calculating on the fly.  Like this:

```suggestion
    const uppercap = ES.SystemUTCEpochNanoSeconds().plus(YEARNANOS);
```

Reply:
Done, at least for defining DAY_NANOS and using that elsewhere. I'm sure there's a tradeoff between initializing all these bigint values once at the beginning vs using them per-call-site, and I suppose in an ideal world the bundlers / optimizers would be able to streamline this as much as possible (e.g. Closure Compiler can optimize some operations on numbers so the resulting bundle is just the value already initialized, but unsure if without native bigint usage that is feasible here / whether that is a thing other OSS tools do).

Reply:
Cool, sounds good.